pkg/controllers/crds/crds_controller.go

```diff
@@ -438,6 +447,42 @@ func (r *Reconciler) CreateCACert(begin, end time.Time) (*KeyPairArtifacts, erro
 	return &KeyPairArtifacts{Cert: cert, Key: key, CertPEM: certPEM, KeyPEM: keyPEM}, nil
 }
 
+func (r *Reconciler) CreateCAChain(ca *KeyPairArtifacts, begin, end time.Time) (*KeyPairArtifacts, error) {
```

Comment:
This function is the same as the one above with the only exception that it's using `r.CAChainName` instead of `r.CAName`. Please refactor `CreateCACert` to accept two strings and then call them accordingly.

I wonder why the duplication check didn't catch that. 🤔 


Also, why you're here. There is a function calling Reconcile with a lock but it's _not_ using a pointer receiver. :O 

I wonder how that ever worked. It's `checkCRDs` and `checkEndpoints`. Could you please fix them to use a pointer receiver instead? Thanks. :)

Reply:
ah, you're right. I missed the parameters for `der, err := x509.CreateCertificate(rand.Reader, templ, templ, key.Public(), key)`. Well, if you can without making it way more complex than it's worth, then okay. Otherwise it's fine.

Reply:
pointer receiver fixed.

Reply:
Thanks for trying. 😊